src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -125,10 +134,23 @@ public ParseStatus parseJavadocAsDetailNode(DetailAST javadocCommentAst) {
             result.setTree(tree);
         }
         catch (ParseCancellationException | IllegalArgumentException ex) {
-            // If syntax error occurs then message is printed by error listener
-            // and parser throws this runtime exception to stop parsing.
-            // Just stop processing current Javadoc comment.
-            ParseErrorMessage parseErrorMessage = errorListener.getErrorMessage();
+            ParseErrorMessage parseErrorMessage = null;
+
+            if (ex.getCause() instanceof FailedPredicateException
+                    || ex.getCause() instanceof NoViableAltException) {
```

Comment:
I see test which check messages for containing `no viable alternative`, but do they also cover `FailedPredicateException`?